pkg/api/dataproxy.go

```diff
@@ -115,12 +118,28 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 		}
 	}
 
+	outputToAuditLog(ds.Type, c)
+
 	proxy := NewReverseProxy(ds, proxyPath, targetUrl)
 	proxy.Transport, err = ds.GetHttpTransport()
 	if err != nil {
 		c.JsonApiErr(400, "Unable to load TLS certificate", err)
 		return
 	}
+
 	proxy.ServeHTTP(c.Resp, c.Req.Request)
 	c.Resp.Header().Del("Set-Cookie")
 }
+
+func outputToAuditLog(dataSourceType string, c *middleware.Context) {
+	auditLogger := log.New("data-proxy-audit", "userid", c.UserId, "orgid", c.OrgId, "username", c.Login)
+
+	var body string
+	if c.Req.Request.Body != nil {
+		buffer, _ := ioutil.ReadAll(c.Req.Request.Body)
+		c.Req.Request.Body = ioutil.NopCloser(bytes.NewBuffer(buffer))
+		body = string(buffer)
+	}
+
+	auditLogger.Info("Proxying incoming request", "datasource", dataSourceType, "uri", c.Req.RequestURI, "method", c.Req.Request.Method, "body", body)
```

Comment:
Kind of expected the request. Which config section do you think would be appropriate? I could add `enable_audit_log` boolean in the logging section. Offhand it seems like I would need to get a reference to the config here in data proxy for that setting. This might be a bit strange though as then there is a logging setting not handled in log. Any preferences here? Or other pointers that would be useful would be appreciated to help get this resolved quickly, seems like there are a few options.